### PR TITLE
Add Minute_Of_Hour Function As An Alias Of Minute Function

### PR DESCRIPTION
**Pull request by GabeFernandez310**
### Description
Adds support for the `minute_of_hour` function as an alias for the `minute` function which currently exists in opensearch

### Issues Resolved
[#722](https://github.com/opensearch-project/sql/issues/722)
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [X] New functionality has user manual doc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: sql/src/main/antlr/OpenSearchSQLParser.g4 [hunk 1]
```diff
@@ -247,6 +247,7 @@ datetimeConstantLiteral
     | DAY_OF_YEAR
     | LOCALTIME
     | LOCALTIMESTAMP
+    | MINUTE_OF_HOUR
```
Comment (Yury-Fridlyand) [inline sql/src/main/antlr/OpenSearchSQLParser.g4:250]: Same as https://github.com/Bit-Quill/opensearch-project-sql/pull/195#discussion_r1052713305

Comment (GabeFernandez310) [reply sql/src/main/antlr/OpenSearchSQLParser.g4]: Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/196/commits/eb4c99e61b2af5497bcc26e2cc216a021271ec4e

**Review by GumpacG — APPROVED** [review]

File: docs/user/dql/functions.rst [hunk 2]
```diff
@@ -1747,21 +1747,21 @@ Description
 >>>>>>>>>>>
 
 Usage: minute(time) returns the minute for time, in the range 0 to 59.
+The `minute_of_hour` function is provided as an alias.
```
Comment (acarbonetto) [inline docs/user/dql/functions.rst:1750]: underscore for a link

**Review by acarbonetto — APPROVED** [review]

File: core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java [hunk 3]
```diff
@@ -899,7 +902,8 @@ private ExprValue exprMicrosecond(ExprValue time) {
    * @return ExprValue.
    */
   private ExprValue exprMinute(ExprValue time) {
-    return new ExprIntegerValue(time.timeValue().getMinute());
+    return new ExprIntegerValue(
+        (MINUTES.between(LocalTime.MIN, time.timeValue()) % 60));
```
Comment (Yury-Fridlyand) [inline core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java:906]: Does `time.timeValue()` work when first arg is `STRING`?

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java [hunk 4]
```diff
@@ -691,6 +691,62 @@ public void minute() {
     assertEquals("minute(\"2020-08-17 01:02:03\")", expression.toString());
   }
 
+  private void testMinuteOfHour(FunctionExpression dateExpression, int minute, String testExpr) {
+    assertEquals(INTEGER, dateExpression.type());
```
Comment (Yury-Fridlyand) [inline core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java:695]: `assertAll`?

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java [hunk 5]
```diff
@@ -691,6 +691,62 @@ public void minute() {
...
+  private void testInvalidMinuteOfHour(String time) {
+    FunctionExpression expression = DSL.minute_of_hour(DSL.literal(new ExprTimeValue(time)));
+    eval(expression);
+  }
+
+  @Test
+  public void minuteOfHourInvalidArguments() {
+    when(nullRef.type()).thenReturn(TIME);
+    when(missingRef.type()).thenReturn(TIME);
+    assertEquals(nullValue(), eval(DSL.minute_of_hour(nullRef)));
```
Comment (Yury-Fridlyand) [inline core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java:733]: `assertAll`?

File: docs/user/dql/functions.rst (hunk 2 above)
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:1750]:
```suggestion
The `minute_of_hour`_ function is provided as an alias.
```

File: docs/user/dql/functions.rst [hunk 6]
```diff
@@ -1747,21 +1747,21 @@ Description
...
 
 Argument type: STRING/TIME/DATETIME/TIMESTAMP
 
 Return type: INTEGER
 
 Example::
 
-    os> SELECT MINUTE((TIME '01:02:03'))
+    os> SELECT MINUTE((TIME '01:02:03')), MINUTE_OF_HOUR((TIME '01:02:03'))
```
Comment (Yury-Fridlyand) [inline docs/user/dql/functions.rst:1758]: Does `TIME '01:02:03'` work? I think it should be `TIME('01:02:03')` or just `'01:02:03'`.

File: docs/user/dql/functions.rst (hunk 2 above)
Comment (Yury-Fridlyand) [reply docs/user/dql/functions.rst:1750]: We have no section for `minute_of_hour`

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java (hunk 4 above)
Comment (GabeFernandez310) [reply core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java]: Fixed in https://github.com/Bit-Quill/opensearch-project-sql/pull/196/commits/b3ee7b44118577aa72156856809b533931f81289

File: core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java (hunk 5 above)
Comment (GabeFernandez310) [reply core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java]: Fixed in 17510afbbb5ae86cf1df6e209d0726b2bed35735

File: docs/user/dql/functions.rst (hunk 6 above)
Comment (GabeFernandez310) [reply docs/user/dql/functions.rst]: Fixed in 6deaafb541257764eae4ec96409144b04ba455bd

**Review by Yury-Fridlyand — APPROVED** [review]

**Pull request merged**